src/commons/authentication/oidc.rs

```diff
@@ -0,0 +1,368 @@
+    }
+
+    /// Returns the path of the files containing client id and secret in case they are given.
+    pub fn client_credentials_mount_paths(secret_name: &str) -> (String, String) {
+        let volume_mount_path = Self::client_credentials_volume_mount_path(secret_name);
+
+        (
+            format!("{volume_mount_path}/{CLIENT_ID_SECRET_KEY}"),
+            format!("{volume_mount_path}/{CLIENT_SECRET_SECRET_KEY}"),
+        )
+    }
+
+    /// Name of the clientId and clientSecret env variables.
+    ///
+    /// Env variables need to be C_IDENTIFIER according to k8s docs. We could replace `-` with `_` and `.` with `_`,
+    /// but this could cause collisions. To be collision-free we hash the secret key instead and use the hash in the env var.
+    pub fn client_credentials_env_names(secret_name: &str) -> (String, String) {
+        let mut hasher = DefaultHasher::new();
+        secret_name.hash(&mut hasher);
+        let secret_name_hash = hasher.finish();
+
+        // Prefix with zeros to have consistent length. Max length is 16 characters, which is caused by [`u64::MAX`].
+        let secret_name_hash = format!("{:016x}", secret_name_hash).to_uppercase();
+        let env_var_prefix = format!("OIDC_{secret_name_hash}");
+
+        (
+            format!("{env_var_prefix}_CLIENT_ID"),
+            format!("{env_var_prefix}_CLIENT_SECRET"),
+        )
+    }
+
+    pub fn client_credentials_env_var_mounts(secret_name: String) -> Vec<EnvVar> {
+        let (client_id_env_var, client_secret_env_var) =
+            Self::client_credentials_env_names(&secret_name);
+
+        vec![
+            EnvVar {
+                name: client_id_env_var,
+                value_from: Some(EnvVarSource {
+                    secret_key_ref: Some(SecretKeySelector {
+                        key: CLIENT_ID_SECRET_KEY.to_string(),
+                        name: Some(secret_name.clone()),
+                        ..Default::default()
+                    }),
+                    ..Default::default()
+                }),
+                ..Default::default()
+            },
+            EnvVar {
+                name: client_secret_env_var,
+                value_from: Some(EnvVarSource {
+                    secret_key_ref: Some(SecretKeySelector {
+                        key: CLIENT_SECRET_SECRET_KEY.to_string(),
+                        name: Some(secret_name),
+                        ..Default::default()
+                    }),
+                    ..Default::default()
+                }),
+                ..Default::default()
+            },
+        ]
+    }
+}
+
+/// An enum of supported OIDC or identity providers which can serve as a hint
+/// in the product operator. Some products require special handling of
+/// authentication related config options. This hint can be used to enable such
+/// special handling.
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "PascalCase")]
```

Comment:
I feel like we should switch that to `lowercase` or `camelCase`.

Reply:
I think there is a naming convention of using PascalCase for scalar values in k8s. However I can't find it :/

Reply:
Alright, then let's keep it as is.